### ISSUE TITLE
Add an additional step between scanning QR code and adding to CovPass to mitigate accidental use of CovPass instead of CovPassCheck

### ISSUE THREAD

**Issue by jkrwdf** [Digitaler-Impfnachweis/covpass-android#103]
## Avoid duplicates
* [X] This enhancement request has not already been raised before

This enhancement request affects Android and iOS. 
According to https://github.com/Digitaler-Impfnachweis/documentation/issues/27 I shall proceed here with this sentence. 

## Current Implementation
CovPass could be mistakenly used instead of CovPassCheck at locations of DCC verification. 

It is very simple to confuse both, and as Linus Neumann from the CCC reported in one of the last episodes of "Logbuch Netzpolitik", he was already asked by someone who verifies certificates why at the end of the day all those certificates must be manually deleted from the verifier-phone ;-)

I already made a suggestion for a technical solution to this issue in https://github.com/corona-warn-app/cwa-wishlist/issues/601 (for the CWA) which was rejected, however a quick-win is possible nevertheless which I present here. 

## Suggested Enhancement
After scanning a DCC, an explicit dialog shall appear with this question: 

> You have scanned a digital COVID certificate. Do you want to add it now to this app? Note: If you are right now acting as person verifying certificates of other people, you are in the wrong App here and should use CovPassCheck. 

(sentence can of course be adapted for wording and style, important is the idea).

This step shall not have a "Do not show again" function. 

## Expected Benefits
Yes, this is an annoying additional step for everyone who uses CovPass in the intended way. 

But it will quite unambiguously inform people who accidentally use CovPass instead of CovPassCheck.

**Comment by timokoenig** [comment]
More or less a duplicate of https://github.com/Digitaler-Impfnachweis/covpass-android/issues/99 but I like the additional step to add the certificate to your device. This in combination with a better scan overlay in CovPassCheck would mitigate the issue

**Comment by Ein-Tim** [comment]
@jkrwdf 

Feel free to open the same / an adjusted version of this issue in the cwa-wishlist repository with a reference to https://github.com/corona-warn-app/cwa-wishlist/issues/666. 

As https://github.com/corona-warn-app/cwa-wishlist/issues/601 was a different idea, I'm sure a new issue which presents the idea you raised here would not get rejected (if it get's implemented is another question).

**Comment by jkrwdf** [comment]
@Ein-Tim: Will consider. 

In the meantime I have prepared a demonstration what I would imagine:

![image](https://user-images.githubusercontent.com/5329491/141644985-cef7b035-f4f7-4471-b758-5314be8b4202.png)

**Comment by alexcimander** [comment]
thank you all for the discussion! 
Having an extra "notice" before importing the certificate is a great idea, also adding a different overlay. We are currently discussing those approaches and will keep you updated about it. So please bear with us 🙏

**Comment by Ein-Tim** [comment]
FYI: There is a related discussion in https://github.com/ehn-dcc-development/hcert-spec/issues/107 about this.

**Comment by Ein-Tim** [comment]
I made a concrete suggestion for improving this in https://github.com/Digitaler-Impfnachweis/covpass-ios/issues/71.

**Comment by jleufgen** [comment]
@jkrwdf can we then close this ticket if there is another ticket on this topic now?

**Comment by jkrwdf** [comment]
Hello @jleufgen ,

you want to close my ticket (opened November 12) because @Ein-Tim has opened another ticket on November 30 for the other platform (which is, according to your FAQ, see https://github.com/Digitaler-Impfnachweis/documentation/issues/27, redundant as you have no common issue tracker for issues addressing both platforms)?

Or do you want to close it because on https://github.com/Digitaler-Impfnachweis/certification-apis/issues/206 there is a new discussion about emergency measures?

I personally consider the idea proposed in this issue still relevant for consideration by the app architects.

JKr

**Comment by Ein-Tim** [comment]
Transfering a comment from @alexcimander from https://github.com/Digitaler-Impfnachweis/covpass-ios/issues/71#issuecomment-993514898 to this issue:

> we are currently working on multiple features that tackle the misuse of the CovPass as a verifier-app. A screen asking if the import is actually wanted is one of them. We are planning to release it with the next release. We will keep you updated about it! Thank you for the great ideas 👍 🎆

**Comment by jkrwdf** [comment]
With the recent Corona-Warn-App update, it implemented this popup when certificates for three people are already stored and you add the fourth, fifth, ... person:

![image](https://user-images.githubusercontent.com/5329491/147049915-22367569-8079-4a07-8d33-18c7cd3803f6.png)

**Comment by jkrwdf** [comment]
With version 1.15.0 this dialogue appears when 9 persons have their certificates in CovPass and the 10th person's certificate has been scanned:

![Screenshot_20220114-201655](https://user-images.githubusercontent.com/5329491/149572872-a071b7c8-5d90-4daf-8807-88a022cfb267.png)

Adding is still possible, but the motivation of this issue has been addressed.

**Comment by timokoenig** [comment]
It is better but why is it not shown with every certificate that is being scanned? Now you have to make a mistake 10 times to get notified 😄 imo that should appear earlier

**Comment by Ein-Tim** [comment]
@timokoenig IHMO It should appear if 3 certificates are already stored in the app. This is how CWA handles it.

**Comment by timokoenig** [comment]
@Ein-Tim is it 3 certificates or 3 different people?

**Comment by Ein-Tim** [comment]
@timokoenig 3 different people, sorry.

**Comment by timokoenig** [comment]
Ok that's much better. For every new person or at least after three different people is better than 10. @alexcimander what was the reason behind this magical number?

**Comment by timokoenig** [comment]
Just looked at the iOS implementation and they show the view for person 2 and 10. I guess Android will do it the same. It doesn't really make sense for me

https://github.com/Digitaler-Impfnachweis/covpass-ios/blob/8b25bcfcdc0f108f8c74179a09e5ae1849f2b344/Source/CovPassCommon/Sources/CovPassCommon/Repository/VaccinationRepository.swift#L267

**Comment by jkrwdf** [comment]
That's affirmative. Did not see this because I already had two people in my wallet before starting my test.

**Comment by pucherd** [comment]
Hello @jkrwdf,
Yes, the limitation is displayed at 2 and 10 persons scan and once we reach max 20 persons we block storing certificates in CovPass.

**Comment by ingowerren** [comment]
In 2021, the European Union introduced the digital COVID certificate as an EU-wide recognised proof of vaccination, recovery or negative testing for COVID-19. This regulation expired on 30 June 2023. On a transitional basis, the EU will continue to operate the technical systems to enable cross-border certificate checks until 31 December 2023. The transitional operation will end on that date.
In Germany, the possibility of issuing digital COVID certificates such as vaccination and recovery certificates will therefore be discontinued on 31 December 2023. The CovPass app and the CovPassCheck app will therefore also be discontinued. The CovPass app will be switched to wallet mode, where the stored certificates will be retained.

What does this mean for you?
All functionalities will continue to beavailable until 31 December 2023.

**From 1 January 2024:**

Support for the app will no longer be offered.

- The apps will no longer be available for (re)download in the app store.
- No new functions will be published.
- Further information can also be found in the CovPass app privacy policy.
